apps/examples/board_specific_demo/ameba_spi_lcd_demo/example_spi_lcd.c

```diff
@@ -0,0 +1,152 @@
+static pid_t g_task = 0;
+
+void test_put_area()
+{
+	int fd = 0, p = 0, count = 0;
+	char port[20] = {'\0'};
+	sprintf(port, LCD_DEV_PATH, p);
+	fd = open(port, O_RDWR | O_SYNC, 0666);
+	if (fd < 0) {
+		printf("ERROR: Failed to open lcd port : %s error:%d\n", port, fd);
+		return -1;
+	}
+	struct lcddev_area_s area;
+	area.row_start = 10;
+	area.row_end = 19;
+	area.col_start = 10;
+	area.col_end = 19;
+	area.stride = 10*2;
+	area.planeno = 0;
+	int i=0;
+	uint8_t spi_data[401];
+	for(i=0;i<=100;i+=2)
+	{
+  		spi_data[i] = 0x07E0 & 0X00FF;
+  		spi_data[i+1] = (0x07E0 & 0xFF00)>>8; 
+	}
+	for(i=100;i<=200;i+=2)
+	{
+		spi_data[i] = 0x001F & 0X00FF;
+  		spi_data[i+1] = (0x001F & 0xFF00)>>8; 
+	}
+	for(i=200;i<=400;i+=2)
+	{
+		spi_data[i] = 0xF800 & 0X00FF;
+  		spi_data[i+1] = (0xF800 & 0xFF00)>>8; 
+	}
+	area.data = &spi_data;
+	ioctl(fd, LCDDEVIO_PUTAREA,  (unsigned long)(uintptr_t)&area);
+	sleep(1);
+	uint8_t *spi_data_big = (uint8_t *)malloc(ST7789_XRES*ST7789_YRES*2+1);
+	area.row_start = 0;
+	area.row_end = ST7789_YRES - 1;
+	area.col_start = 0;
+	area.col_end = ST7789_XRES - 1;
+	area.stride = 2*ST7789_XRES;
+	for(i=0;i<ST7789_XRES*ST7789_YRES*2;i+=2)
+	{
+		spi_data_big[i] = 0xF8;
+		spi_data_big[i+1] = 0x00;
+	}
+	area.data = spi_data_big;
+	ioctl(fd, LCDDEVIO_PUTAREA,  (unsigned long)(uintptr_t)&area);
+	sleep(2);
+	area.row_start = ST7789_YRES/2-30;
+	area.row_end = ST7789_YRES/2+30;
+	area.col_start = ST7789_XRES/2-30;
+	area.col_end = ST7789_XRES/2+30;
+	for(i=0;i<ST7789_XRES*ST7789_YRES*2;i+=2)
+	{
+		spi_data_big[i] = 0xFF;
+		spi_data_big[i+1] = 0xFF;
+	}
+	ioctl(fd, LCDDEVIO_PUTAREA,  (unsigned long)(uintptr_t)&area);
+	close(fd);
+}
+void test_put_run()
+{
+	int fd = 0, p = 0, count = 0;
+	char port[20] = {'\0'};
+	sprintf(port, LCD_DEV_PATH, p);
+	fd = open(port, O_RDWR | O_SYNC, 0666);
+	if (fd < 0) {
+		printf("ERROR: Failed to open lcd port : %s error:%d\n", port, fd);
+		return -1;
+	}
+	struct lcddev_run_s run;
+	run.planeno = 0;
+	run.col = 60;
+	run.row = 150;
+	run.npixels = 100;
+	int i=0;
+	uint8_t spi_data[201];
+	spi_data[1] = 0x07E0 & 0X00FF;
+  	spi_data[0] = (0x07E0 & 0xFF00)>>8; 
+	run.data = &spi_data;
+	for(i=0;i<=200;i+=2)
+	{
+  		spi_data[i+1] = 0xF800 & 0X00FF;
+  		spi_data[i] = (0xF800 & 0xFF00)>>8; 
+	}
+	ioctl(fd, LCDDEVIO_PUTRUN,  (unsigned long)(uintptr_t)&run);
+	close(fd);
+}
+void test_clear()
+{
+	int ret, fd = 0, p = 0;
+	char port[20] = {'\0'};
+	sprintf(port, LCD_DEV_PATH, p);
+	fd = open(port, O_RDWR | O_SYNC, 0666);
+	if (fd < 0) {
+		printf("ERROR: Failed to open lcd port : %s error:%d\n", port, fd);
+		return -1;
+	}
+	ioctl(fd, LCDDEVIO_CLEAR, &ret);
+	close(fd);
+}
+
+void SPIDemoShow_task(void)
+{
+	int count = 0;
+	while(count<5)
+	{
+		test_clear();
+		sleep(1);
+		test_put_run();
+		sleep(1);
+		test_put_area();
+		count++;
+		sleep(2);
+	}
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int ameba_spi_lcd_main(int argc, char *argv[])
```

Comment:
Is there anything related with amebasmart? If not, let's locate this under example, not in board_specific_demo.